doc/SONiC_OC_NTP_HLD.md

```diff
@@ -0,0 +1,491 @@
+# Feature Name
+NTP Support in Management Framework
+
+# High Level Design Document
+
+#### Rev 0.1
+
+# Table of Contents
+  * [List of Tables](#list-of-tables)
+  * [Revision](#revision)
+  * [About This Manual](#about-this-manual)
+  * [Scope](#scope)
+  * [Definition/Abbreviation](#definitionabbreviation)
+
+# List of Tables
+[Table 1: Abbreviations](#table-1-abbreviations)
+
+# Revision
+| Rev |     Date    |       Author       | Change Description                |
+|:---:|:-----------:|:------------------:|-----------------------------------|
+| 0.1 | 05/03 |   Bing Sun         | Initial version                   |
+  
+  
+# About this Manual
+
+This document introduces the support of NTP configuration using management framework. It also describes the mechanism of ntp service upon various NTP configuration changes.
+
+# Scope
+
+This document covers NTP "configuration" and "show" commands based on the OpenConfig YANG model, as well as the backend mechanism required to support each command.
+NTP unit tests are also included.
+
+# Definition/Abbreviation
+
+### Table 1: Abbreviations
+| **Term**                 | **Meaning**                         |
+|--------------------------|-------------------------------------|
+|          NTP             |  Network Time Protocol              |
+|          ntpd            |  NTP Daemon                         |
+
+
+# 1 Feature Overview
+
+NTP stands for Network Time Protocol. It is used to synchronize the time of a computer or server to another server or refernce time source.   
+
+SONiC click CLI provides commands to add/delete the IP address of a remote NTP server. Multiple NTP servers can be configured, and both IPv4 an IPv6 are supported. SONiC click CLI also provides the show command to display status of NTP peers.    
+
+With this feature, users will be provided the same capabilities via Management CLI, REST and gNMI using OpenConfig Yang models.
+It also provides users to configure NTP source and NTP vrf (Management VRF for now?) for addtional ntp control.
+
+## 1.1 Requirements
+
+### 1.1.1 Front end configuration and get capabilities
+
+#### 1.1.1.1 add/delete NTP server      
+This installs NTP server information in the Redis ConfigDB and in the ntp.conf file (NTP configuration file).   
+The NTP server can be IPv4 address, IPv6 address , or the name of a NTP server.   
+Mutliple NTP servers can be configured.
+  
+#### 1.1.1.2 add NTP source
+This installs the global NTP source information in the Redis ConfigDB and in the ntp.conf file. It enables ntpd to listen to the specific source.      
+NTP source can be configured as either IP address or interface name.    
+Only one NTP source will be installed as the global NTP source. A new configured NTP source will override the existing NTP source.   

```

Comment:
There is no confirmation for data VRF yet.  But yes, both data VRF and source interface should be able to configured.

Reply:
So, when we listen to only the source interface, we will not get packets from any other interface right ? Can we document this, somewhere and call it out. Otherwise, this might be a surprise to folks assuming that we listen on the default-VRF therefore we should receive packets on any interface.

Reply:
What about 127.0.0.1?. NTP need to listen on this aslo with source-interface as this is used for internal communication with ntpd.

Reply:
Source-interface does not mean it has to route packets via that interface. Source interface means it has to use given interface ip as source-ip of packets while sending ntp request. For example one can specify Loopback as source interface and packets can be sent out via eth0.
If NTP listens on source interface then this will fail.

Reply:
NTP will still listen on 127.0.0.1, for ntpq to work.